dev/AppNotifications/AppNotificationTelemetry.h

```diff
@@ -0,0 +1,319 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include <WindowsAppRuntimeInsights.h>
+#include <wrl\wrappers\corewrappers.h>
+
+DECLARE_TRACELOGGING_CLASS(AppNotificationTelemetryProvider,
+    "Microsoft.WindowsAppSDK.Notifications.AppNotificationTelemetry",
+    // {1825c850-a487-537d-b768-f0ab298d2565}
+    (0x1825c850, 0xa487, 0x537d, 0xb7, 0x68, 0xf0, 0xab, 0x29, 0x8d, 0x25, 0x65));
+
+
+class AppNotificationTelemetry : public wil::TraceLoggingProvider
+{
+    IMPLEMENT_TELEMETRY_CLASS(AppNotificationTelemetry, AppNotificationTelemetryProvider);
+
+public:
+    DEFINE_EVENT_METHOD(RegisterByAPI)(
+        winrt::hresult hr,
+        std::wstring const& appId) noexcept try
+    {
+        if (c_maxEventLimit >= UpdateLogEventCount())
```

Comment:
@brflynn Throttling being implemented here using a count pattern. Thoughts on this Vs the approach you implemented recently?

Reply:
That should be good enough for basic telemetry. That's the same pattern that push utilizes. We can revisit when we take on the tasks for comprehensive telemetry